### ISSUE TITLE
Unable to connect my OSM account

### ISSUE THREAD

**Issue by jbelien** [westnordost/StreetComplete#1597]
It seems impossible to connect my osm account. The "loading" modal seems to be running indefinitely.

**How to Reproduce**

When I try to connect my OSM account, I can open the osm.org OAUth window in the browser (Firefox), grant access to StreetComplete but when it goes back to StreetComplete, the "Loading" screen seems to run indefinitely.

**Versions affected**

Android 9
StreetComplete 14.1
Firefox 68.1.1

**Comment by westnordost** [comment]
You go back to StreetComplete, or are you sent back to StreetComplete?

**Comment by jbelien** [comment]
> You go back to StreetComplete, or are you sent back to StreetComplete?

When I click on "Grant access" on the osm.org OAuth page, the StreetComplete app opens again on the settings page with a loading "screen" (loading in a white square on top on the settings page) and that loading screen never goes away.

**Comment by jbelien** [comment]
Oh well ... I just tried again and it worked !
I'm now connected to my OSM account.

Maybe it was just an issue with osm.org OAuth system.
Sorry for the noise !

--- 

Maybe it would be nice to have some kind of "error" message if there is an issue (or timeout) with the osm.org OAuth.

**Comment by westnordost** [comment]
Hmm weird. The loading indicator does not actually wait for any process to complete, but it waits for Firefox to call back to StreetComplete with the correct parameters. So, there can not be a timeout.

**Comment by jbelien** [comment]
Maybe have a look at how [Vespucci](https://github.com/MarcusWolschon/osmeditor4android) do it.
I connected my account to both app at the same time and didn't have an issue with Vespucci.
The only difference I noticed is that Vespucci open the osm.org OAuth page directly in the app, compared to StreetComplete that request to open the osm.org OAuth page in an external navigator (and in my case, I configured my phone to be able to select the navigator each time it tries to open a new link - that's maybe what caused the issue I had).
